src/parsec.rs

```diff
@@ -224,3 +320,100 @@ fn new_rng() -> Box<dyn Rng> {
         Box::new(rng)
     }
 }
+
+#[cfg(all(test, feature = "mock_parsec"))]
+mod tests {
+    use super::*;
+    use crate::{chain::SectionInfo, routing_table::Prefix, xor_name::XorName};
+    use unwrap::unwrap;
+
+    const DEFAULT_MIN_SECTION_SIZE: usize = 4;
+
+    #[test]
+    fn parsec_size_counter() {
+        let mut counter = ParsecSizeCounter::default();
+        assert!(!counter.needs_pruning());
+        counter.increase_size(PARSEC_SIZE_LIMIT);
+        assert!(!counter.needs_pruning());
+        counter.increase_size(1);
+        assert!(counter.needs_pruning());
+    }
+
+    fn create_full_ids() -> Vec<FullId> {
+        (0..DEFAULT_MIN_SECTION_SIZE)
+            .map(|_| FullId::new())
+            .collect()
+    }
+
+    fn create_gen_pfx_info(full_ids: Vec<FullId>, version: u64) -> GenesisPfxInfo {
+        let members = full_ids.iter().map(|id| *id.public_id()).collect();
+        let section_info = unwrap!(SectionInfo::new_for_test(
+            members,
+            Prefix::<XorName>::default(),
+            version
+        ));
+        GenesisPfxInfo {
+            first_info: section_info,
+            first_state_serialized: Vec::new(),
+            latest_info: SectionInfo::default(),
+        }
+    }
+
+    // Create a ParsecMap with 3 sections
+    fn create_parsec_map() -> ParsecMap {
+        let log_ident = LogIdent::new("node");
+        let full_ids = create_full_ids();
+        let full_id = full_ids[0].clone();
+
+        let gen_pfx_info = create_gen_pfx_info(full_ids.clone(), 0);
+        let mut parsec_map = ParsecMap::new(full_id.clone(), &gen_pfx_info);
+
+        let gen_pfx_info = create_gen_pfx_info(full_ids.clone(), 1);
+        parsec_map.init(full_id.clone(), &gen_pfx_info, &log_ident);
+
+        let gen_pfx_info = create_gen_pfx_info(full_ids, 2);
+        parsec_map.init(full_id, &gen_pfx_info, &log_ident);
+
+        parsec_map
+    }
+
+    fn parsec_map_to_vec(map: BTreeMap<u64, ParsecAndCounter>) -> Vec<(u64, ParsecSizeCounter)> {
+        map.into_iter()
+            .map(|(id, ParsecAndCounter { size_counter, .. })| (id, size_counter))
+            .collect()
+    }
+
+    #[test]
+    fn parsec_map_check_prune_required() {
+        let response = Response::new();
+        let full_id = FullId::new();
+        let pub_id = full_id.public_id();
+
+        let log_ident = LogIdent::new("node");
+        let mut parsec_map = create_parsec_map();
+
+        // First handle responses with parsec 1 which should not trigger a need for pruning.
+        for _ in 0..=PARSEC_SIZE_LIMIT {
+            assert!(!parsec_map.handle_response(1, response.clone(), *pub_id, &log_ident));
```

Comment:
Should we send responses and requests to test both?

Reply:
Yes that sounds like a good idea